datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -887,6 +887,17 @@ pub async fn from_substrait_rel(
                         not_impl_err!("Union relation requires at least one input")
                     }
                 }
+                set_rel::SetOp::IntersectionPrimary => {
+                    if set.inputs.len() == 2 {
+                        LogicalPlanBuilder::intersect(
+                            from_substrait_rel(ctx, &set.inputs[0], extensions).await?,
+                            from_substrait_rel(ctx, &set.inputs[1], extensions).await?,
+                            false,
+                        )
+                    } else {
+                        not_impl_err!("Primary Intersect relation with more than two inputs isn't supported")
```

Comment:
If we associate this DataFusion operation with `IntersectionMultiset` instead of `IntersectionPrimary`, we could potentially handle multiple inputs by chaining intersects. For example
```
SetRel(
  op = SET_OP_INTERSECTION_MULTISET,
  inputs = [<A>, <B>, <C>]
)
```
becomes something like
```rust
LogicalPlanBuilder::intersect(
  from_substrait_rel(ctx, &set.inputs[0], extensions).await?,
  LogicalPlanBuilder::intersect(
    from_substrait_rel(ctx, &set.inputs[1], extensions).await?,
    from_substrait_rel(ctx, &set.inputs[2], extensions).await?,
    false
  )
false)
```